hoi/simulation/simulation_hois.py

```diff
@@ -0,0 +1,758 @@
+        time_bump (None): Time bump parameter. Default is None.
+        time_length_bump (None): Time length bump parameter. Default is None.
+        triplet_character (list): List of triplet characteristics.
+        Default is None.
+        triplet_character_with_beh (list): List of triplet characteristics with
+        behavioral information. Default is None.
+
+    Returns:
+        list: A list containing the simulated data, simulated data
+        without behavioral information over time, and simulated
+        data with behavioral information over time.
+
+    """
+
+    n_triplets = int(n_nodes / 3)
+
+    if time_length_bump is None:
+        time_length_bump = [
+            int(n_times / n_triplets) for i in range(n_triplets)
+        ]
+
+    if time_bump is None:
+        time_bump = []
+        cc = int(time_length_bump[0] / 2)
+        time_bump.append(cc)
+        for tt in time_length_bump:
+            cc = cc + tt
+            time_bump.append(cc)
+
+    mean_mvgauss, cov_al = cov_all_beh(
+        n_nodes=n_nodes,
+        triplet_character=triplet_character,
+        triplet_character_with_beh=triplet_character_with_beh,
+    )
+
+    han_list = []
+    for i, t_length in enumerate(time_length_bump):
+        aa = np.zeros(n_times)
+        aa[
+            time_bump[i]
+            - int(t_length / 2) : time_bump[i]
+            - int(t_length / 2)
+            + t_length
+        ] = np.hanning(t_length)
+        aa_temp = np.tile(
+            aa[np.newaxis, np.newaxis, :], (n_trials, n_nodes, 1)
+        )
+        han_list.append(aa_temp)
+
+    han_inverse_list = []
+    for i, t_length in enumerate(time_length_bump):
+        aa = np.ones(n_times)
+        aa[
+            time_bump[i]
+            - int(t_length / 2) : time_bump[i]
+            - int(t_length / 2)
+            + t_length
+        ] = 1 - np.hanning(t_length)
+        aa_temp = np.tile(
+            aa[np.newaxis, np.newaxis, :], (n_trials, n_nodes, 1)
+        )
+        han_inverse_list.append(aa_temp)
+
+    sim_static_rand = np.random.multivariate_normal(
+        mean_mvgauss,
+        cov=np.identity(n_nodes + 1),
+        size=n_trials,
+        check_valid="warn",
+        tol=1e-8,
+    )
+    sim_rand = np.tile(sim_static_rand[:, :, np.newaxis], (1, 1, n_times))
+
+    sim_static_hois = np.random.multivariate_normal(
+        mean_mvgauss, cov=cov_al, size=n_trials, check_valid="warn", tol=1e-8
+    )
+    sim_hois = np.tile(sim_static_hois[:, :, np.newaxis], (1, 1, n_times))
+
+    for i, aaa in enumerate(zip(han_list, han_inverse_list)):
+        han, han_inv = aaa
+        sim_rand[:, i * 3 : (i + 1) * 3, :] = (
+            sim_hois[:, i * 3 : (i + 1) * 3, :]
+            * han[:, i * 3 : (i + 1) * 3, :]
+            + sim_rand[:, i * 3 : (i + 1) * 3, :]
+            * han_inv[:, i * 3 : (i + 1) * 3, :]
+        )
+
+    simulated_data = sim_rand
+
+    return [simulated_data[:, :12, :], sim_hois[:, 12, 0]]
```

Comment:
Output list is not necessary. Just `simulated_data[:, :12, :], sim_hois[:, 12, 0]`. Btw, it's a bit weird to have hard-coded variables like `12`...